internal/provider/provider.go

```diff
@@ -416,14 +416,13 @@ func providerConfigure(p *schema.Provider) schema.ConfigureContextFunc {
 		if !skipProviderRegistration {
 			// List all the available providers and their registration state to avoid unnecessary
 			// requests. This also lets us check if the provider credentials are correct.
-			providerList, err := client.Resource.ProvidersClient.List(ctx, nil, "")
+			availableResourceProviders, err := resourceproviders.CacheSupportedProviders(ctx, client.Resource.ProvidersClient)
```

Comment:
we should use the same approach we do for Cached Locations here?

Reply:
Got it, but the location cache is part of the `go-azure-helper`, I can make another PR for it, but I think it will be a breaking change of `go-azure-helper`.

Reply:
Hi @katbyte , not sure I followed, do you mean that we should move the resource provider cache logic to `go-azure-helper`?

Reply:
yes? as tom suggested we should follow the same approach we used for location

Reply:
The move-cache-to-go-azure-helper has been made here: https://github.com/hashicorp/go-azure-helpers/pull/158. Please help review that and after that appreciate ms-henglu could continue here. @tombuildsstuff @katbyte